mart/attack/perturber/perturber.py

```diff
@@ -59,6 +62,22 @@ def projector_wrapper(perturber_module, args):
         if projector is not None:
             self.register_forward_pre_hook(projector_wrapper)
 
+    def on_run_start(self, adversary, input, target, model, **kwargs):
+        self.initialize_parameters(input, target)
+
+    def initialize_parameters(self, input, target):
```

Comment:
I would just move the body of `initialize_parameter` into `on_run_start`.

Reply:
Done